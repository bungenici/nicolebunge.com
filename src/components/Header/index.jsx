import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Header extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
  };

  static defaultProps = {
    children: null,
    className: '',
  };

  render() {
    const { children, className, ...otherProps } = this.props;

    return (
      <header className={classNames(className, styles.header)} role="banner" {...otherProps}>
        {children}
      </header>
    );
  }
}

export default Header;
