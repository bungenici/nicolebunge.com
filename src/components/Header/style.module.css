.header {
  bottom: 0;
  box-shadow: 2em 0 2em -2em rgba(0, 0, 0, 0.125);
  display: flex;
  flex-direction: column;
  height: 100vh;
  left: 0;
  min-width: 12em;
  padding: 1.5em;
  position: sticky;
  top: 0;
  width: 20vw;
}
