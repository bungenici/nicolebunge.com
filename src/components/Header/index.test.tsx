/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Header from '.';

describe('Header component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
