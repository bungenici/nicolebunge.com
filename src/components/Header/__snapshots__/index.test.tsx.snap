// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
    role="banner"
  />
</DocumentFragment>
`;
